### ISSUE TITLE
In code, I think "t -> t+1" is correct. can you explain it ?

### ISSUE THREAD

**Issue by taki0112** [NVlabs/denoising-diffusion-gan#11]
Hello.

I read your code and your paper (in Figure 3), and I got confused.
![image](https://user-images.githubusercontent.com/26713992/178696792-7e682317-1a06-4140-9998-21e07f1c5b0d.png)

So, I think it is correct. (from [link](https://github.com/NVlabs/denoising-diffusion-gan/blob/6818ded6443e10ab69c7864745457ce391d4d883/train_ddgan.py#L342) in train_ddgan.py)
**Because of `x_tp1` got from `t+1`.**

```
def q_sample(coeff, x_start, t, *, noise=None):
    if noise is None:
      noise = torch.randn_like(x_start)
      
    x_t = extract(coeff.a_s_cum, t, x_start.shape) * x_start + \
          extract(coeff.sigmas_cum, t, x_start.shape) * noise
    
    return x_t

def q_sample_pairs(coeff, x_start, t):
    noise = torch.randn_like(x_start)
    x_t = q_sample(coeff, x_start, t)
    x_t_plus_one = extract(coeff.a_s, t+1, x_start.shape) * x_t + \
                   extract(coeff.sigmas, t+1, x_start.shape) * noise
    
    return x_t, x_t_plus_one
```

* Line 378: `D_real = netD(x_t, t, x_tp1.detach()).view(-1)` **->** D_real = netD(x_t, **t+1,** x_tp1.detach()).view(-1)
* Line 342: `x_0_predict = netG(x_tp1.detach(), t, latent_z)` **->** x_0_predict = netG(x_tp1.detach(), **t+1**, latent_z)
* Line 381: `output = netD(x_pos_sample, t, x_tp1.detach()).view(-1)` **->** output = netD(x_pos_sample, **t+1**, x_tp1.detach()).view(-1)
* Line 378: `x_0_predict = netG(x_tp1.detach(), t, latent_z)` **->** x_0_predict = netG(x_tp1.detach(), **t+1**, latent_z)
* Line 379: ` x_pos_sample = sample_posterior(pos_coeff, x_0_predict, x_tp1, t)` **->**  x_pos_sample = sample_posterior(pos_coeff, x_0_predict, x_tp1, **t+1**)
* Line 381: `output = netD(x_pos_sample, t, x_tp1.detach()).view(-1)` **->** output = netD(x_pos_sample, **t+1**, x_tp1.detach()).view(-1)
* Line 411~414: 
```
x_0_predict = netG(x_tp1.detach(), t, latent_z)
x_pos_sample = sample_posterior(pos_coeff, x_0_predict, x_tp1, t)
output = netD(x_pos_sample, t, x_tp1.detach()).view(-1)
```
**->**
x_0_predict = netG(x_tp1.detach(), **t+1**, latent_z)
x_pos_sample = sample_posterior(pos_coeff, x_0_predict, x_tp1, **t+1**)
output = netD(x_pos_sample, **t+1**, x_tp1.detach()).view(-1)

I could be wrong. Can you explain which one is correct?

**Comment by junyang-zhou** [comment]
Same confusion. It is different from the equations showed in Figure 3 in paper. 

But I guess maybe no matter 't' or 't+1', the result won't change much.

**Comment by blackstarcrash** [comment]
Same confusion.
